src/ldp/http/BasicRequestParser.ts

```diff
@@ -38,7 +38,7 @@ export class BasicRequestParser extends RequestParser {
     }
     const target = await this.targetExtractor.handleSafe(request);
     const preferences = await this.preferenceParser.handleSafe(request);
-    const metadata = await this.metadataExtractor.handleSafe(request);
+    const metadata = await this.metadataExtractor.handleSafe({ request, identifier: target });
```

Comment:
I don't like this. I think `metadataExtractor` should just receive `request` for consistency, and __internally__ use `targetExtractor`. So:


```suggestion
    const metadata = await this.metadataExtractor.handleSafe(request);
    const target = metadata.identifier;
```

Reply:
There's a communication issue here I think since what you suggest here is how I interpreted https://github.com/solid/community-server/pull/438#discussion_r553163529 , but then below that we discussed not to do that. Unless you mean both calling the target extractor in the request parser and also calling it in the metadata extractor?

Reply:
I see I phrased that comment confusingly. I meant:
- Let's have (only) `this.metadataExtractor` extract the target
- Let's have `BasicRequestParser` read it from there
- Let's return it in the operation anyway

So like you've written it now—but do you agree with that approach?

Reply:
That's how I interpreted the comment I linked so my answer is what I said there 😄 . No strong objection, just wondering if it might make the entire parsing process/configuration a bit less readable (since the target extractor is more hidden away).

Reply:
That's the drawback; but it's just the single class then here. It doesn't leak to others.

Reply:
Looking into this now. `metadata.identifier` is a `NamedNode` though, not a `ResourceIdentifier`. So should we ever want to include extra metadata in the `ResourceIdentifier` this could become problematic (we already talked about including the origin at some point to handle acl origin checks). 

There are ways around this: a helper function that converts a metadata object to an identifier by extracting the correct metadata, have the metadata extractor return both metadata and identifier, or store an identifier internally in the metadata and have some way to get it back. But all of these would require additional changes.

Reply:
Grrr, it's not easy. But I definitely prefer the consistency of having `request` as a param everywhere.

Then back to everything having its own `TargetExtractor` I presume.

Reply:
We already broke having `request` as param everywhere when we added `metadata` as param for the body parser anyway.

But so keep it as it is in this PR then? I don't think we want a target extractor in both the request parser and the metadata extractor.

Reply:
I wonder if they just need `target` and `headers`. Can't seem to find a good reason to send the entire request through.

But let's at least be consistent in another way then and:
1. have all of the others receive `{ request }` such that we can extend later as needed
2. perhaps change `identifier` to `target` (such that it is clear what kind of identifier it is)

We can do 1 in another PR if you want.

And I think we also want to do that here https://github.com/solid/community-server/blob/61aa2e12bddf3fd2be8f6265750d26c09e5e24a9/src/ldp/AuthenticatedLdpHandler.ts#L118-L122

and send the `target` along to `credentialsExtractor` as well.